vmaas_sync/metrics_cyndi_test.go

```diff
@@ -0,0 +1,88 @@
+package vmaas_sync //nolint:golint,stylecheck
+
+import (
+	"app/base/core"
+	"app/base/database"
+	"app/base/utils"
+	"github.com/stretchr/testify/assert"
+	"testing"
+	"time"
+)
+
+func TestCyndiSystemsCounts(t *testing.T) {
+	utils.SkipWithoutDB(t)
+	core.SetupTestEnvironment()
+
+	counts, err := getCyndiCounts(refTime())
+	assert.Nil(t, err)
+	assert.Equal(t, 16, counts[lastUploadLast1D])
+	assert.Equal(t, 16, counts[lastUploadAll])
+}
+
+func TestUploadedCyndiSystemsCounts1D(t *testing.T) {
+	utils.SkipWithoutDB(t)
+	core.SetupTestEnvironment()
+
+	systemsQuery := database.Db.Table("inventory.hosts")
+	systemsQuery = updateCyndiQueryLastUpload(systemsQuery, refTime(), 1)
+	var nSystems int
+	assert.Nil(t, systemsQuery.Count(&nSystems).Error)
+	assert.Equal(t, 16, nSystems)
+}
+
+func TestUploadedCyndiSystemsCounts7D(t *testing.T) {
+	utils.SkipWithoutDB(t)
+	core.SetupTestEnvironment()
+
+	systemsQuery := database.Db.Table("inventory.hosts")
+	systemsQuery = updateCyndiQueryLastUpload(systemsQuery, refTime(), 7)
+	var nSystems int
+	assert.Nil(t, systemsQuery.Count(&nSystems).Error)
+	assert.Equal(t, 16, nSystems)
+}
+
+func TestUploadedCyndiSystemsCounts30D(t *testing.T) {
+	utils.SkipWithoutDB(t)
+	core.SetupTestEnvironment()
+
+	systemsQuery := database.Db.Table("inventory.hosts")
+	systemsQuery = updateCyndiQueryLastUpload(systemsQuery, refTime(), 30)
+	var nSystems int
+	assert.Nil(t, systemsQuery.Count(&nSystems).Error)
+	assert.Equal(t, 16, nSystems)
+}
+
+func TestUploadedCyndiSystemsCountsNoSystems(t *testing.T) {
+	utils.SkipWithoutDB(t)
+	core.SetupTestEnvironment()
+
+	systemsQuery := database.Db.Table("inventory.hosts")
+	refTime := time.Date(2020, 9, 23, 10, 0, 0, 0, time.UTC)
+	systemsQuery = updateCyndiQueryLastUpload(systemsQuery, refTime, 30)
+	var nSystems int
+	assert.Nil(t, systemsQuery.Count(&nSystems).Error)
+	assert.Equal(t, 0, nSystems)
+}
+
+func TestUploadedCyndiSystemsCountsAllSystems(t *testing.T) {
+	utils.SkipWithoutDB(t)
+	core.SetupTestEnvironment()
+
+	systemsQuery := database.Db.Table("inventory.hosts")
+	systemsQuery = updateCyndiQueryLastUpload(systemsQuery, refTime(), -1)
+	var nSystems int
+	assert.Nil(t, systemsQuery.Count(&nSystems).Error)
+	assert.Equal(t, 16, nSystems)
+}
+
+func TestCyndiTags(t *testing.T) {
+	utils.SkipWithoutDB(t)
+	core.SetupTestEnvironment()
+
+	cyndiData, err := getCyndiData()
+	assert.Nil(t, err)
+	assert.Equal(t, 16, cyndiData.SystemsCount)
+	assert.Equal(t, 2, cyndiData.UniqueTags)
```

Comment:
Why two unique tags? I see more unique tags in testing data:
~~~~
{"key": "k1", "value": "val1", "namespace": "ns1"}
{"key": "k2", "value": "val2", "namespace": "ns1"}
{"key": "k3", "value": "val4", "namespace": "ns1"}
{"key": "k3", "value": "val5", "namespace": "ns1"}
~~~~
.. so at least 4, if I count correctly.